rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/CatalogRequestBuilder.java

```diff
@@ -0,0 +1,118 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.server.serialization;
+
+import static org.ow2.proactive_grid_cloud_portal.rm.shared.CatalogConstants.*;
+
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.UUID;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.mime.HttpMultipartMode;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+
+
+public class CatalogRequestBuilder {
+
+    private final CatalogObjectAction catalogObjectAction;
+
+    public CatalogRequestBuilder(CatalogObjectAction catalogObjectAction) {
+        this.catalogObjectAction = catalogObjectAction;
+    }
+
+    public String build() {
+        StringBuilder builder = new StringBuilder(URL_CATALOG).append("/buckets/")
+                                                              .append(this.catalogObjectAction.getBucketName());
+
+        if (this.catalogObjectAction.isRevision()) {
+            builder.append("/resources/").append(this.catalogObjectAction.getNodeSourceName()).append("/revisions?");
+            appendParameters(builder, paramPair(COMMIT_MESSAGE_PARAM, this.catalogObjectAction.getCommitMessage()));
+        } else {
+            builder.append("/resources?");
+            appendParameters(builder,
+                             paramPair(NAME_PARAM, this.catalogObjectAction.getNodeSourceName()),
+                             paramPair(KIND_PARAM, this.catalogObjectAction.getKind()),
+                             paramPair(COMMIT_MESSAGE_PARAM, this.catalogObjectAction.getCommitMessage()),
+                             paramPair(OBJECT_CONTENT_TYPE_PARAM, this.catalogObjectAction.getObjectContentType()));
+        }
+        return builder.toString();
+    }
+
+    private void appendParameters(StringBuilder builder, String... paramPairs) {
+        Arrays.stream(paramPairs).forEach(paramPair -> builder.append(paramPair).append("&"));
+        builder.deleteCharAt(builder.lastIndexOf("&"));
+    }
+
+    public CloseableHttpResponse postNodeSourceRequestToCatalog(String sessionId, String fullUri,
+            HttpServletResponse response)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)